src/main/java/ca/corefacility/bioinformatics/irida/service/impl/ProjectSubscriptionServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+package ca.corefacility.bioinformatics.irida.service.impl;
+
+import ca.corefacility.bioinformatics.irida.exceptions.EntityNotFoundException;
+import ca.corefacility.bioinformatics.irida.model.subscription.ProjectSubscription;
+import ca.corefacility.bioinformatics.irida.model.user.User;
+import ca.corefacility.bioinformatics.irida.repositories.ProjectSubscriptionRepository;
+import ca.corefacility.bioinformatics.irida.service.ProjectSubscriptionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Service;
+
+import javax.validation.Validator;
+
+/**
+ * A specialized service layer for project subscriptions.
+ */
+@Service
+public class ProjectSubscriptionServiceImpl extends CRUDServiceImpl<Long, ProjectSubscription>
+        implements ProjectSubscriptionService {
+
+    private final ProjectSubscriptionRepository projectSubscriptionRepository;
```

Comment:
Can you fix the formatting of this file, it should be indented with tabs and not spaces. Might just need to update your ide to set tabs as default for .java files.

Reply:
Please see 242aeb0522f0fe6211596050c7f3a2dcc6ef7b32.